sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c

```diff
@@ -354,25 +441,40 @@ static struct snd_soc_dai_link_component platform_component[] = {
 
 static void rt711_init(const struct snd_soc_acpi_link_adr *link,
 		       struct snd_soc_dai_link *dai_links,
+		       struct codec_info *info,
 		       bool playback)
 {
+	/*
+	 * headset should be initialized once.
+	 * Do it with dai link for playback.
+	 */
 	if (!playback)
 		return;
 
 	dai_links->init = headset_init;
 }
 
-static int rt1308_num;
 static void rt1308_init(const struct snd_soc_acpi_link_adr *link,
 			struct snd_soc_dai_link *dai_links,
+			struct codec_info *info,
 			bool playback)
 {
-	if (!playback)
-		return;
-
-	rt1308_num++;
-	if (rt1308_num > 1)
-		dai_links->init = second_spk_init;
+	info->count++;
+	if (info->count == 1)
```

Comment:
info->amp_num would be nicer for reviewers, for consistency with dmic_num for example.

Reply:
done